server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/TypeNameParserFactory.scala

```diff
@@ -0,0 +1,13 @@
+package org.apache.james.jmap.change
+
+import javax.inject.Inject
+
+class TypeNameParserFactory @Inject() (setTypeName: Set[TypeName]) {
```

Comment:
Injects needs to be performed on java.util.Set not to confuse Guice. You need then to convert it to a scala set.

Reply:
Can we have unit tests for this class?